documentation/proposals/Proposal - Multi-Backend Input.md

```diff
@@ -0,0 +1,184 @@
+- The `IInputBackend.Activate` and `IInputBackend.Deactivate` methods must be called by the `InputContext.Add` and `InputContext.Remove` methods (respectively)
+
+**KEY POINT FOR WORKING GROUP**: The Windowing-Input integration mandates the use of source generators. Is this ok?
+
+# Proposed API
+
+```diff
+namespace Silk.NET.Input
+{
+-   public interface IInputPlatform
+-   {
+-       bool IsApplicable(IView view);
+-       IInputContext CreateInput(IView view);
+-   }
+}
+```
+
+```diff
+namespace Silk.NET.Input
+{
++   /// <summary>
++   /// Encapsulates input devices sourced from input backends.
++   /// </summary>
+-   public interface IInputContext : IDisposable
++   public class InputContext : IDisposable
+    {
+-       nint Handle { get; }
+        IReadOnlyList<IGamepad> Gamepads { get; }
+        IReadOnlyList<IJoystick> Joysticks { get; }
+        IReadOnlyList<IKeyboard> Keyboards { get; }
+        IReadOnlyList<IMouse> Mice { get; }
+        IReadOnlyList<IInputDevice> OtherDevices { get; }
++
++       /// <summary>
++       /// Gets a list of backends from which all input devices on this input context are sourced.
++       /// </summary>
++       IReadOnlyList<IInputBackend> Backends { get; }
+        event Action<IInputDevice, bool>? ConnectionChanged;
++
++       /// <summary>
++       /// Adds the given backend to the list of <see cref="Backends" />.
++       /// </summary>
++       void Add(IInputBackend backend);
++
++       /// <summary>
++       /// Removes the given backend from the list of <see cref="Backends" />.
++       /// </summary>
++       void Remove(IInputBackend backend);
++
++       /// <summary>
++       /// Updates all input data on all backends.
++       /// </summary>
++       void Update();
+    }
+}
+```
+
+```diff
+namespace Silk.NET.Input
+{
++   /// <summary>
++   /// Represents an input backend from which input devices can be retrieved.
++   /// </summary>
++   public interface IInputBackend
++   {
++       /// <summary>
++       /// Gets all connected devices recognised by this backend.
++       /// </summary>
++       IReadOnlyList<IInputDevice> ConnectedDevices { get; }
++
++       /// <summary>
++       /// Raised when an input device is connected or disconnected.
++       /// </summary>
++       event Action<IInputDevice, bool>? ConnectionChanged;
++
++       /// <summary>
++       /// Updates this backend's input data.
++       /// </summary>
++       /// <remarks>
++       /// Input events and changes are permitted to occur outside of this method.
++       /// </remarks>
++       void Update();
++
++       /// <summary>
++       /// Activates this input backend and runs any necessary prerequisites.
++       /// </summary>
++       /// <remarks>
++       /// Called by <see cref="InputContext.Add" />
++       /// </remarks>
++       void Activate();
++
++       /// <summary>
++       /// Deactivates this input backend and tears down any resources created.
```

Comment:
> tears down any resources created
why not `Dispose` + `IDisposable`?
Is this supposed to be de- and re-activated at will? That seems like a pointless complication.
Why not activate -> ctor and deactivate -> `Dispose`, the cost of recreating an instance seems very low compared to how often this should be done.

Reply:
It is intended that backends can be passed between contexts. Will update docs to indicate my intentions.

Reply:
Updated docs.